### PR TITLE
[main][feat]: deploy nftStaking script

### PR DESCRIPTION
**Pull request by abc-alpaca**
## Description
- deployment script for nftStaking contract
- add new key "NFT" in json and add "NFTStaking"

## Why?
- to test NftStaking contract

## ChangeLogs:
- remove constructor when deploy
- add script deploy with writing json file

### CODE REVIEW

**Comment by spicysquid168** [comment]
The structure in config file should just be
```
NFTStaking: "someAddress"
```

**Review by alpanaca — DISMISSED** [review]

File: deploy/exec/nft-staking/deploy/nft-staking.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  ░██╗░░░░░░░██╗░█████╗░██████╗░███╗░░██╗██╗███╗░░██╗░██████╗░
+  ░██║░░██╗░░██║██╔══██╗██╔══██╗████╗░██║██║████╗░██║██╔════╝░
+  ░╚██╗████╗██╔╝███████║██████╔╝██╔██╗██║██║██╔██╗██║██║░░██╗░
+  ░░████╔═████║░██╔══██║██╔══██╗██║╚████║██║██║╚████║██║░░╚██╗
+  ░░╚██╔╝░╚██╔╝░██║░░██║██║░░██║██║░╚███║██║██║░╚███║╚██████╔╝
+  ░░░╚═╝░░░╚═╝░░╚═╝░░╚═╝╚═╝░░╚═╝╚═╝░░╚══╝╚═╝╚═╝░░╚══╝░╚═════╝░
+  Check all variables below before execute the deployment script
+  */
+
+  console.log(">> Deploying an upgradable NFTStaking contract");
```
Comment (spicysquid168) [inline deploy/exec/nft-staking/deploy/nft-staking.ts:19]: No need log here. `UpgradeableContractDeployer` takes care of that already.

File: solidity/contracts/8.13/nft-staking/NFTStaking.sol [hunk 2]
```diff
@@ -97,8 +97,6 @@ contract NFTStaking is INFTStaking, OwnableUpgradeable, ReentrancyGuardUpgradeab
     _;
   }
 
-  constructor() initializer {}
```
Comment (spicysquid168) [inline solidity/contracts/8.13/nft-staking/NFTStaking.sol:100]: I think can leave this as is for security reason.

**Review by alpanaca — DISMISSED** [review]

File: package.json [hunk 3]
```diff
@@ -120,6 +120,7 @@
     "deploy:mainnet:AVController": "hardhat --network mainnet deploy --config hardhat.config.ts --no-compile --reset --tags AVController",
     "deploy:mainnet:config:set-params:AVController": "hardhat --network mainnet deploy --config hardhat.config.ts --no-compile --reset --tags AVControllerSetParams",
     "deploy:mainnet:delta-neutral-vault-config:set-controller": "hardhat --network mainnet deploy --config hardhat.config.ts --no-compile --reset --tags DeltaNeutralVaultConfigSetController",
+    "deploy:mainnet:nft-staking:deploy:nft-staking": "hardhat --network mainnet deploy --config hardhat.config.ts --no-compile --reset --tags NFTStaking",
     "deploy:testnet:000": "hardhat --network testnet deploy --no-compile --reset --tags Testnet",
```
Comment (spicysquid168) [inline package.json:124]: add `deploy:mainnet:nft-staking:config:add-pool`

**Review by spicysquid168 — APPROVED** [review]

**Review by alpanaca — APPROVED** [review]

**Pull request merged**